datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -688,6 +688,20 @@ fn build_predicate_expression(
                 return Ok(unhandled);
             }
         }
+        Expr::InList {
+            expr,
+            list,
+            negated,
+        } if !list.is_empty() && list.len() < 20 => {
+            let eq_fun = if *negated { Expr::not_eq } else { Expr::eq };
+            let re_fun = if *negated { Expr::and } else { Expr::or };
+            let change_expr = list
+                .iter()
+                .map(|e| eq_fun(*expr.clone(), e.clone()))
+                .reduce(re_fun)
```

Comment:
MAP / REDUCE[1] -- love it!

[1] https://static.googleusercontent.com/media/research.google.com/en//archive/mapreduce-osdi04.pdf